ecosystem/sf-indexer/aptos-sf-indexer/src/models/tokens.rs

```diff
@@ -0,0 +1,213 @@
+    pub royalty_points_numerator: bigdecimal::BigDecimal,
+    pub royalty_points_denominator: bigdecimal::BigDecimal,
+    pub maximum_mutable: bool,
+    pub uri_mutable: bool,
+    pub description_mutable: bool,
+    pub properties_mutable: bool,
+    pub royalty_mutable: bool,
+    pub default_properties: serde_json::Value,
+    // Default time columns
+    pub inserted_at: chrono::NaiveDateTime,
+}
+
+impl TokenData {
+    fn from_token_data(token_data: &TokenDataPB) -> Self {
+        let token_data_id = token_data.token_data_id.as_ref().unwrap().clone();
+        Self {
+            creator_address: token_data_id.creator_address,
+            collection_name: token_data_id.collection_name,
+            name: token_data_id.name,
+            transaction_version: token_data.transaction_version as i64,
+            maximum: u64_to_bigdecimal(token_data.maximum),
+            supply: u64_to_bigdecimal(token_data.supply),
+            largest_property_version: u64_to_bigdecimal(token_data.largest_property_version),
+            metadata_uri: token_data.metadata_uri.clone(),
+            royalty_points_numerator: u64_to_bigdecimal(token_data.royalty_points_denominator),
+            royalty_points_denominator: u64_to_bigdecimal(token_data.royalty_points_denominator),
+            maximum_mutable: token_data.maximum_mutable,
+            uri_mutable: token_data.uri_mutable,
+            description_mutable: token_data.description_mutable,
+            properties_mutable: token_data.properties_mutable,
+            royalty_mutable: token_data.royalty_mutable,
+            default_properties: serde_json::to_value(&token_data.default_properties).unwrap(),
+            inserted_at: chrono::Utc::now().naive_utc(),
+        }
+    }
+}
+
+#[derive(Debug, FieldCount, Identifiable, Insertable, Queryable, Serialize)]
+#[primary_key(
+    creator_address,
+    collection_name,
+    name,
+    property_version,
+    transaction_version,
+    owner_address
+)]
+#[diesel(table_name = "token_ownership")]
+pub struct TokenOwnership {
+    pub creator_address: String,
+    pub collection_name: String,
+    pub name: String,
+    pub property_version: bigdecimal::BigDecimal,
+    pub transaction_version: i64,
+    pub owner_address: String,
+    pub amount: bigdecimal::BigDecimal,
+    // Default time columns
+    pub inserted_at: chrono::NaiveDateTime,
+}
+
+impl TokenOwnership {
+    fn from_token(token: &TokenPB) -> Self {
+        let token_data_id = token
+            .token_id
+            .as_ref()
+            .unwrap()
+            .token_data_id
+            .as_ref()
+            .unwrap()
+            .clone();
+        Self {
+            creator_address: token_data_id.creator_address,
+            collection_name: token_data_id.collection_name,
+            name: token_data_id.name,
+            property_version: u64_to_bigdecimal(token.token_id.as_ref().unwrap().property_version),
+            transaction_version: token.transaction_version as i64,
```

Comment:
with tnx_version, to get the owned amount seems requires one query to get the largest version and another query to get corresponding amount now. Is this intented?

Reply:
ah yes. we'll have a current version of this table. This table is for transaction history.